src/docdb/getCosmosClient.ts

```diff
@@ -12,6 +12,8 @@ import { ext } from "../extensionVariables";
 export function getCosmosClient(endpoint: string, key: string, isEmulator: boolean | undefined): CosmosClient {
 
     const vscodeStrictSSL: boolean | undefined = vscode.workspace.getConfiguration().get<boolean>(ext.settingsKeys.vsCode.proxyStrictSSL);
-    return new CosmosClient({ endpoint, key, userAgentSuffix: appendExtensionUserAgent(), agent: new https.Agent({ rejectUnauthorized: isEmulator ? !isEmulator : vscodeStrictSSL }) });
+    const enableEndpointDiscovery: boolean | undefined = vscode.workspace.getConfiguration().get<boolean>(ext.settingsKeys.enableEndpointDiscovery);
+    const connectionPolicy = { enableEndpointDiscovery: (enableEndpointDiscovery === undefined) ? true : enableEndpointDiscovery };
```

Comment:
Sorry, I'm not super familiar with this, but from this issue https://github.com/microsoft/vscode-cosmosdb/issues/1857, my understanding is that we want the user to be able to set this to be `false`, correct?  And the default behavior is that it is usually set to `true`?  

In the documentation linked in the issue and in the SDK index.d.ts, `enableEndpointDiscovery` is an optional property, so I would have assumed that it would default to false.  But I just want to make sure that the point of this setting is to allow users to set this to `false` if their private endpoint requires them to.